finding/finding.go

```diff
@@ -52,45 +55,142 @@ type Location struct {
+	OutcomeNotAvailable
+	_
+	_
+	_
+	// OutcomeError indicates an errors while running.
+	// The results could not be determined.
+	OutcomeError
+	_
+	_
+	_
 	// OutcomePositive indicates a positive outcome.
 	OutcomePositive
-	// OutcomeNotApplicable indicates a non-applicable outcome.
-	OutcomeNotApplicable
+	_
+	_
+	_
 	// OutcomeNotSupported indicates a non-supported outcome.
 	OutcomeNotSupported
 )
 
 // Finding represents a finding.
 // nolint: govet
 type Finding struct {
-	Rule        string            `json:"rule"`
-	Outcome     Outcome           `json:"outcome"`
-	Risk        rule.Risk         `json:"risk"`
-	Message     string            `json:"message"`
-	Location    *Location         `json:"location,omitempty"`
-	Remediation *rule.Remediation `json:"remediation,omitempty"`
+	Probe       string             `json:"probe"`
+	Outcome     Outcome            `json:"outcome"`
+	Message     string             `json:"message"`
+	Location    *Location          `json:"location,omitempty"`
+	Remediation *probe.Remediation `json:"remediation,omitempty"`
 }
 
-// New creates a new finding.
-func New(loc embed.FS, ruleID string) (*Finding, error) {
-	r, err := rule.New(loc, ruleID)
+// AnonymousFinding is a finding without a corerpsonding probe ID.
+type AnonymousFinding struct {
+	Finding
+	// Remove the probe ID from
+	// the structure until the probes are GA.
+	Probe string `json:"probe,omitempty"`
+}
+
+var errInvalid = errors.New("invalid")
+
+// FromBytes creates a finding for a probe given its config file's content.
+func FromBytes(content []byte, probeID string) (*Finding, error) {
+	p, err := probe.FromBytes(content, probeID)
 	if err != nil {
 		// nolint
 		return nil, err
 	}
 	f := &Finding{
-		Rule:        ruleID,
+		Probe:       p.Name,
+		Outcome:     OutcomeNegative,
+		Remediation: p.Remediation,
+	}
+	return f, nil
+}
+
+// New creates a new finding.
+func New(loc embed.FS, probeID string) (*Finding, error) {
+	p, err := probe.New(loc, probeID)
+	if err != nil {
+		return nil, fmt.Errorf("%w", err)
+	}
+
+	f := &Finding{
+		Probe:       p.Name,
 		Outcome:     OutcomeNegative,
-		Remediation: r.Remediation,
+		Remediation: p.Remediation,
+	}
+	return f, nil
+}
+
+// NewWith create a finding with the desried location and outcome.
+func NewWith(efs embed.FS, probeID, text string, loc *Location,
+	o Outcome,
+) (*Finding, error) {
+	f, err := New(efs, probeID)
+	if err != nil {
+		return nil, fmt.Errorf("finding.New: %w", err)
+	}
+
+	f = f.WithMessage(text).WithOutcome(o).WithLocation(loc)
+	return f, nil
+}
+
+// NewWith create a negative finding with the desried location.
+func NewNegative(efs embed.FS, probeID, text string, loc *Location,
+) (*Finding, error) {
+	f, err := NewWith(efs, probeID, text, loc, OutcomeNegative)
+	if err != nil {
+		return nil, fmt.Errorf("finding.NewWith: %w", err)
 	}
-	if r.Remediation != nil {
-		f.Risk = r.Risk
+
+	f = f.WithMessage(text).WithLocation(loc)
```

Comment:
what does this extra line do? isn't location set by `NewWith`?

do you need anything more in these functions than 
```
return NewWith(efs, probeID, text, loc, OutcomeNegative)
```

Reply:
same question applies for all of these NewOutcome functions

Reply:
+1, must be a copy-paste mistake. Fixed.

Reply:
i dont think this is resolved. 

We don't need `f = f.WithMessage(text)` either as NewWith takes care of the text. The same applies to all NewFoo functions.

Reply:
ho you're right! Fixed.